include/service/entities/plugin_service.js

```diff
@@ -1202,63 +1241,63 @@ module.exports = function PluginServiceModule(pb) {
+            },
 
-             //process localization
-             function(callback) {
-
-                 self.getLocalizations(plugin.dirName, function(err, localizations) {
-                     if (util.isNullOrUndefined(localizations)) {
-                         return callback(null, true);
-                     }
-                     
-                     for (var locale in localizations) {
-                         pb.log.debug('PluginService:[%s] Registering localizations for locale [%s]', details.uid, locale);
-
-                         var opts = {
-                             site: self.site,
-                             plugin: details.uid
-                         };
-                         var result = pb.Localization.registerLocale(locale, localizations[locale], opts);
-                         if (!result && pb.log.isDebug()) {
-                             pb.log.debug('PluginService:[%s] Failed to register localizations for locale [%s].  Is the locale supported in your configuration?', details.uid, locale);
-                         }
-                     }
-                     callback(null, !util.isError(err) && result);
-                 });
-             }
-        ];
-        async.series(tasks, function(err, results) {
-            //cleanup on error
-            if (util.isError(err) && details && details.uid) {
-                delete ACTIVE_PLUGINS[details.uid];
+            //process localization
+            function(callback) {
+
+                self.getLocalizations(plugin.dirName, function(err, localizations) {
+                    for (var locale in localizations) {
+                        if (pb.log.isDebug()) {
+                            pb.log.debug('PluginService:[%s] Registering localizations for locale [%s]', details.uid, locale);
+                        }
+
+                        var opts = {
+                            site: self.site,
+                            plugin: details.uid
+                        };
+                        var result = pb.Localization.registerLocale(locale, localizations[locale], opts);
+                        if (!result && pb.log.isDebug()) {
+                            pb.log.debug('PluginService:[%s] Failed to register localizations for locale [%s].  Is the locale supported in your configuration?', details.uid, locale);
+                        }
+                    }
+                    callback(null, !util.isError(err) && result);
+                });
             }
+        ];
 
-            //callback with final result
-            var success = !util.isError(err);
-            pb.log.info('PluginService: Initialized plugin %s (%s). RESULT=[%s]', plugin.name, plugin.uid, success);
-            cb(err, success);
-        });
+        if(cached_plugin) {
+            details = cached_plugin.details;
+            plugin.dependencies = details.dependencies;
```

Comment:
Since you are returning on the next line do you need to set the details and plugin.dependencies variable?